sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -216,8 +216,16 @@ class Dataset[T] private[sql](
   private[sql] def resolve(colName: String): NamedExpression = {
     queryExecution.analyzed.resolveQuoted(colName, sparkSession.sessionState.analyzer.resolver)
       .getOrElse {
-        throw new AnalysisException(
-          s"""Cannot resolve column name "$colName" among (${schema.fieldNames.mkString(", ")})""")
+        if (schema.fieldNames.contains(colName)) {
+          throw new AnalysisException(
+            s"""Cannot resolve column name "$colName" among (${schema.fieldNames.mkString(", ")}).
+               | Try adding backticks to the column name, i.e., `$colName`"""
```

Comment:
I would explain, for instance, if the name parts in the column should be kept as the part of its column name, try to quote them by backticks.

Reply:
@HyukjinKwon Thanks for the review!

Sorry I don't quite understand your sentence here:
>  if the name parts in the column should be kept as the part of its column name

Would you mind elaborating what do you mean?

Reply:
Ah, I mean if the name parts of the column `a.b.c` should be considered as the name of whole column itself like `` `a.b.c` ``

Reply:
I see, how about:
```
Try adding backticks to the column name, i.e., `$colName`, if $colName is the name of the whole column
```
I am fine with either one

Reply:
Yup, please go ahead.